roboteam_ai/src/utilities/OffensiveCoach.cpp

```diff
@@ -0,0 +1,317 @@
+//
+// Created by robzelluf on 3/8/19.
+//
+
+#include <roboteam_ai/src/interface/widget.h>
+#include <roboteam_ai/src/interface/drawer.h>
+#include "OffensiveCoach.h"
+
+namespace rtt {
+namespace ai {
+namespace coach {
+
+int OffensiveCoach::maxPositions = 4;
+double OffensiveCoach::maxDistanceFromBall = 6.0;
+double OffensiveCoach::newRobotPositionMargin = 0.05;
+double OffensiveCoach::marginFromLines = 0.2;
+std::vector<OffensiveCoach::OffensivePosition> OffensiveCoach::offensivePositions;
+std::map<int, OffensiveCoach::OffensivePosition> OffensiveCoach::robotPositions;
+
+double OffensiveCoach::calculateCloseToGoalScore(Vector2 position) {
+    double distanceFromGoal = (Field::get_their_goal_center() - position).length();
+
+    double score = exp(-0.1 * distanceFromGoal);
+    return score;
+}
+
+double OffensiveCoach::calculateShotAtGoalScore(Vector2 position, roboteam_msgs::World world) {
+    double safelyness = 3;
```

Comment:
Can we use another word like safeDistanceFactor or so? This word just feels wrong.

Reply:
agreed

Reply:
Done